### PR TITLE
HubSync v1.0

### PR DESCRIPTION
**Pull request by TheJaySmith**
This will sync gcr to dockerhub per #1320 . It functions in this way

1) runs gcloud to pull down all image repos within gcr.io/kubeflow-images-public
2) runs a follow-up gcloud command to pull down image tags (latest)
3) Build cloudbuild.yaml file to authenticate with DockerHub, pull the image from gcr.io/kubeflow-images-public, tag the image, and push it to DockerHub

This is base code and is pre-coded largely just to pull from GCR to DockerHub. Users could modify the code in order to push to their own GCR or DockerHub. keys.yaml is the file where you store your credentials. 

I am hoping for a version 2.0 with a CLI. This would break it down into steps so that a user can build and sync what they want. However, for now, this will move all images from gcr.io/kubeflow-images-public to DockerHub

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/1548)

### CODE REVIEW

File: kubeflow/weaveflux/README.md [hunk 1]
```diff
@@ -25,8 +25,17 @@ Finally, we will deploy the application
 ks apply default -c weaveflux
 ```
 
+<<<<<<< HEAD
```
Comment (jlewi) [inline kubeflow/weaveflux/README.md]: Looks like you have some merge conflicts in the file.

**Comment by jlewi** [comment]
Please use "yaml" not "yml" as the extension.

Can you reference in the PR description the issue tracking syncing the Docker images?

Can you expand the PR description? What I'd like to understand is what this PR is doing in terms of the overall plan to sync our images to DockerHub.

It looks like this is a script that uses gcloud to get a list of the images and then creates a YAML file to retag those images using GCB?

**Comment by TheJaySmith** [comment]
Hello, 

Fixed merge conflicts and yaml and updated comment

/retest

**Comment by jlewi** [comment]
Thanks for the really great description. Sorry for the slow reply been busy preparing for the summit and helping with the 0.3 release.

I don't think we need to resolve this as part of this PR, but I thought the primary audience for this tool was the Kubeflow release team to make it easy to sync our images from GCR to DockerHub to support users who can't use GCR. Would we still need a CLI in that case?

Ideally I'd like to automate this. The current thinking is we are using Prow to manage building and pushing our Docker images. Prow has a periodic job which is equivalent to a cron job. So my initial thought is that we would setup a prow periodic job to run this script regularly so that we could sync all our images to DockerHub.

File: releasing/hubsync/hubsync.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+                #print(imgData)
+                    images.append(b)
+        else:
+            #print(image)
+            images.append(image)
+
+
+for item in images:
+    getTags = subprocess.run(["gcloud","--project=kubeflow-images-public","container","images","list-tags",item,"--format=json","--limit=1"], stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    #print(getTags)
```
Comment (jlewi) [inline releasing/hubsync/hubsync.py]: Delete commented out code.

File: releasing/hubsync/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<<<<<<< HEAD
+=======
```
Comment (jlewi) [inline releasing/hubsync/requirements.txt]: Looks like you have some merge conflicts?

**Comment by jlewi** [comment]
It looks like this would push all of our images; even if they already exist in DockerHub. Is this something we will need to fix later?

**Comment by jlewi** [comment]
/test all

Test failure looks like a flake since this PR isn't actually changing any code that would affect kfctl.

**Comment by TheJaySmith** [comment]
@jlewi it should just add an extra image tag in the future. 

My thoughts around a CLI are more for the end user. The script as it exists today takes all images and uploads them to another source. It's possible an end user might want to do some form of version control of the images for their own use or only use certain images for compliance or other policy. The idea behind the CLI would be to make each step it's own command so that an end user can only do what the feel they need.

**Comment by jlewi** [comment]
Looks like the tests are still failing.

**Comment by TheJaySmith** [comment]
@jlewi is there a particular element I can fix? I am having difficulty determining from the log why it's failing.

**Comment by TheJaySmith** [comment]
@jlewi I took a deeper look at the logs and couldn't find anything specific. I may need a second pair of eyes to confirm.

**Comment by jlewi** [comment]
Here' are the pod logs

```
version \"1.10.7-gke.1\" is unsupported.","status":"INVALID_ARGUMENT","statusMessage":"Bad
```

Your GKE deployment is failing because your GKE version is out of date. This is because your code is out of date. You need to sync and then rebase on master to pull in the fix.

**Comment by jlewi** [comment]
@TheJaySmith you've suffered enough.

The test failures are unrelated to this PR since the only code modified is in hubsync which doesn't contain any jsonnet files.

I'll ignore the presubmit failures and merge manually.

**Pull request merged**